pre-commit/app_tests/json_tests.py

```diff
@@ -357,28 +392,34 @@ def check_minimal_data_paths(self):
         """
         Checks to make sure each action includes the minimal required data paths
         """
-        app_json_actions = [
-            act
-            for act in self._app_json["actions"]
-            if act["action"] not in ("test connectivity", "on poll")
-        ]
-        message = None
+        app_json_actions = [act for act in self._app_json["actions"]]
         verbose = []
-        for action in app_json_actions:
+        message = TEST_PASS_MESSAGE
+        for idx, action in enumerate(app_json_actions):
+            if action["action"] in ("test connectivity", "on poll"):
+                continue
             data_paths = set(
                 [
                     (path, data_path_dic["data_type"])
                     for path, data_path_dic in self._get_data_paths(action).items()
                 ]
             )
-            if not MINIMAL_DATA_PATHS.issubset(data_paths):
-                message = "One or more actions are missing a required data path"
-                action_name = action["action"]
-                verbose.append(f"{action_name} is missing one or more required data path")
+            for minimal_path in MINIMAL_DATA_PATHS:
+                if minimal_path not in data_paths:
+                    data_path, data_type = minimal_path
+                    self._app_json["actions"][idx]["output"].append(
+                        {"data_path": data_path, "data_type": data_type}
+                    )
+                    action_name = action["action"]
+                    verbose.append(f"{action_name} missing {data_path}")
```

Comment:
We can flatten this a little bit without sacrificing clarity:
```suggestion
            for missing_path in (MINIMAL_DATA_PATHS - data_paths):
                data_path, data_type = missing_path
                self._app_json["actions"][idx]["output"].append(
                    {"data_path": data_path, "data_type": data_type}
                )
                action_name = action["action"]
                verbose.append(f"{action_name} missing {data_path}")
```